src/celeritas/em/distribution/WentzelDistribution.hh

```diff
@@ -0,0 +1,341 @@
+ */
+CELER_FUNCTION real_type WentzelDistribution::inc_mom_sq() const
+{
+    return inc_energy_ * (inc_energy_ + 2 * inc_mass_);
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Calculate the Moilere screening coefficient.
+ */
+CELER_FUNCTION real_type WentzelDistribution::compute_screening_coefficient() const
+{
+    // TODO: Reference for just proton correction?
+    real_type correction = 1;
+    real_type sq_cbrt_z
+        = fastpow(real_type(target_.atomic_number().get()), real_type{2} / 3);
+    if (target_.atomic_number().get() > 1)
+    {
+        real_type tau = inc_energy_ / inc_mass_;
+        // TODO: Reference for this factor?
+        real_type factor = sqrt(tau / (tau + sq_cbrt_z));
+        real_type inv_beta_sq = 1 + ipow<2>(inc_mass_) / inc_mom_sq();
+
+        correction = min(target_.atomic_number().get() * real_type{1.13},
+                         real_type{1.13}
+                             + real_type{3.76}
+                                   * ipow<2>(target_.atomic_number().get()
+                                             * constants::alpha_fine_structure)
+                                   * inv_beta_sq * factor);
+    }
+
+    return correction * data_.screening_factor * screen_r_sq_elec() * sq_cbrt_z
+           / inc_mom_sq();
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Calculate the (cosine of the) maximum polar angle that incident particle
+ * can scatter off of the target's electrons.
+ */
+CELER_FUNCTION real_type WentzelDistribution::compute_max_electron_cos_t() const
+{
+    // TODO: Need to validate against Geant4 results
+    real_type max_energy = is_electron_ ? real_type{0.5} * inc_energy_
+                                        : inc_energy_;
+    real_type final_energy = inc_energy_ - min(cutoff_energy_, max_energy);
+
+    if (final_energy > 0)
+    {
+        real_type incident_ratio = 1 + 2 * inc_mass_ / inc_energy_;
+        real_type final_ratio = 1 + 2 * inc_mass_ / final_energy;
+        real_type cos_t_max = sqrt(incident_ratio / final_ratio);
+
+        return clamp(cos_t_max, real_type{0}, real_type{1});
+    }
+    else
+    {
+        return 0;
+    }
+}
+//---------------------------------------------------------------------------//
+/*!
+ * Calculate the momentum scale for the nuclear form factor. This should
+ * be a constant function but it has a special case for hydrogen.
+ */
+CELER_FUNCTION real_type WentzelDistribution::nuclear_form_momentum_scale() const
+{
+    // TODO: Geant has a different form momentum scale for hydrogen?
+    if (target_.atomic_number().get() == 1)
+    {
+        return 1 / real_type{3.097e-6};
+    }
+    return native_value_to<MomentumSq>(
+               12
+               * ipow<2>(2 * constants::hbar_planck
+                         / (real_type(1.27e-15) * units::meter)))
+        .value();
```

Comment:
Probably good to note where the 1.27e-15 m comes from, i.e. the approximation for the nuclear radius $r_n = 1.27 A^{0.27}$ fm [LR11] 2.263.

Where does the factor of two come from? Does this work out to be exactly equal to (the inverse of) the hardcoded 6.937e-6 factor in Geant4?

Reply:
I found that citation only recently, so I'll be sure to add it. It comes out to having a factor of 2 difference with the inverse, and I'm in the process of checking where that factor of 2 comes in (section 3 in the overleaf doc). As Soon mentioned earlier, multiplications are preferred so I may invert this in the constexpr function so it more closely matches Geant.

Reply:
```c++
ipow<2>(2 * constants::hbar_planck / (real_type(1.27e-15) * units::meter)))
```
is constexpr so there's no advantage to reducing clarity by inverting it manually.